common/repeat_timer_test.go

```diff
@@ -4,66 +4,86 @@ import (
 	"testing"
 	"time"
 
-	// make govet noshadow happy...
-	asrt "github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/assert"
 )
 
-// NOTE: this only tests with the ManualTicker.
+// NOTE: this only tests with the LogicalTicker.
 // How do you test a real-clock ticker properly?
-func TestRepeat(test *testing.T) {
-	assert := asrt.New(test)
+func TestRepeat(t *testing.T) {
 
 	ch := make(chan time.Time, 100)
-	// tick fires cnt times on ch
+	lt := time.Time{} // zero time is year 1
+
+	// tick fires `cnt` times for each second.
 	tick := func(cnt int) {
 		for i := 0; i < cnt; i++ {
-			ch <- time.Now()
+			lt = lt.Add(time.Second)
+			ch <- lt
 		}
 	}
-	tock := func(test *testing.T, t *RepeatTimer, cnt int) {
+
+	// tock consumes Ticker.Chan() events `cnt` times.
+	tock := func(t *testing.T, rt *RepeatTimer, cnt int) {
 		for i := 0; i < cnt; i++ {
-			after := time.After(time.Second * 2)
+			timeout := time.After(time.Second * 2)
 			select {
-			case <-t.Ch:
-			case <-after:
-				test.Fatal("expected ticker to fire")
+			case _ = <-rt.Chan():
+			case <-timeout:
+				panic("QWE")
+				t.Fatal("expected RepeatTimer to fire")
 			}
 		}
 		done := true
 		select {
-		case <-t.Ch:
+		case <-rt.Chan():
 			done = false
 		default:
 		}
-		assert.True(done)
+		assert.True(t, done)
 	}
 
-	ticker := NewManualTicker(ch)
-	t := NewRepeatTimerWithTicker("bar", ticker)
+	tm := NewLogicalTickerMaker(ch)
+	dur := time.Duration(0) // dontcare
+	rt := NewRepeatTimerWithTickerMaker("bar", dur, tm)
 
-	// start at 0
-	tock(test, t, 0)
+	// Start at 0.
```

Comment:
This sends a time signal to the source, gets received by the logicalTicker, which sets `lasttime`.

In retrospect, what we want to test is not really the specific number of ticks, but the passage of time, to test the various time settings like commit_wait, etc.  So we maybe we need something like this:

```
func TestXYZ(t *testing.T) {
        time := logical.NewTime()
        // It's kinda nice to shadow `time` here.
        time.SetTime("2017-01-01")
        time.Start()
        defer time.Stop()

        node.SetTime(time)
        // test node
    })
}
```
And an environmental variable which could also be passed in via a flag, could set the desired time-interval for testing.  Maybe this could be overridden with time.SetClockInterval().  The only use for it I can think of is to test logical.Time itself, to initialize time.SetClockInterval() globally upon testing, or, to optimize how fast tests run, but maybe that'll never be an issue.